node/Dockerfile

```diff
@@ -0,0 +1,64 @@
+# Miden node Dockerfile
+
+# Setup image builder
+FROM rust:1.76-slim-bookworm AS builder
+
+# Set labels
+LABEL org.opencontainers.image.authors=miden@polygon.io \
+      org.opencontainers.image.url=https://0xpolygonmiden.github.io/ \
+      org.opencontainers.image.documentation=https://github.com/0xPolygonMiden/miden-node \
+      org.opencontainers.image.source=https://github.com/0xPolygonMiden/miden-node \
+      org.opencontainers.image.vendor=Polygon \
+      org.opencontainers.image.licenses=MIT
+
+ARG CREATED
+ARG VERSION
+ARG COMMIT
+LABEL org.opencontainers.image.created=$CREATED \
+      org.opencontainers.image.version=$VERSION \
+      org.opencontainers.image.revision=$COMMIT
+
+# Install dependencies
+RUN apt-get update && \
+    apt-get -y upgrade && \
+    apt-get install -y llvm clang bindgen pkg-config libssl-dev libsqlite3-dev && \
+    rm -rf /var/lib/apt/lists/*
+
+# Copy source code
+WORKDIR /app
+COPY . .
+
+# Cache Cargo dependencies
+RUN --mount=type=cache,target=/usr/local/cargo/registry \
+    cargo fetch --manifest-path node/Cargo.toml
+
+# Build the node crate
+RUN cargo install --features testing --path node
+RUN miden-node make-genesis --inputs-path node/genesis.toml
+
+# Run Miden node
+FROM debian:bookworm-slim
+
+# Update machine & install required packages
+# The instalation of sqlite3 is needed for correct function of the SQLite database
+RUN apt-get update && \
+    apt-get -y upgrade && \
+    apt-get install -y --no-install-recommends \
+    sqlite3 \
+    && rm -rf /var/lib/apt/lists/*
+
+# Define volume for SQLite database files
+VOLUME db
+
+# Copy artifacts from the builder stage
+COPY --from=builder /app/node/miden-node.toml miden-node.toml
```

Comment:
This means we can't change the configuration without having to rebuild all images. This is not good. This should be mounted from the host.

Reply:
I enabled the manual mounting of the file using the `-v` flag.

Reply:
ah, makes sense, so this file is the default, and the user can overwrite it with the flag 👍